accepted/platform-intrinsics.md

```diff
@@ -96,30 +96,70 @@ implementations needs to be available.  As a consequence of this there can be
+- On Intel architecture, there are intrinsics that are only available on float vectors in earlier classes, but are offered for a broader range of types in later classes (e.g. Add over 256-bit vectors on AVX vs. AVX2). For these, it makes sense to explicitly declare the supported types.
+- ARM64 intrinsics include comparisons against zero, but they do not support unsigned types. For these, a trivial JIT expansion could be used to enable all the primitive types to be supported.
+
+In general, it is preferable for a particular instantiated type to not be visible in the API, rather than for the use of an unsupported type to result in a runtime error.
+
+## Versioning and Partial Implementation
+
+The platform intrinsics will be versioned with the runtime. New intrinsics can be added within an existing platform class, but all intrinsics declared in the platform class must be implemented in the JIT at the same time that it is exposed in the library.
+
+## Optionality of Expansion
+
+In general, it is expected that platform intrinsics will be expanded regardless of optimization level. This differs from the handling of other kinds of intrinsics primarily due to the fact that the performance difference between direct execution and fallback (even simply introducing a call) can be quite high, along with the fact that the JIT uses the non-optimizing path for methods that are very large. The combination presents an undesirable performance cliff.
+
+## Supporting Diagnostic Tools
+
+The implementation of the intrinsic must enable indirect invocation (e.g. via diagnostic tools or reflection). This is accomplished as follows:
+   * The source implementation (e.g. in C#) first checks whether the ISA Class is supported, and throws `PlatformNotSupportedException` if not. Otherwise, it performs any necessary checks on the arguments. Finally, if all checks pass, it invokes itself recursively. This call is recursive, and will be expanded by the JIT.
+   * When the JIT encounters an intrinsic that it recognizes, it checks the constraints that it requires to generate the simple expansion. This may include:
+      * Checking whether one or more operands is an immediate. This is required for some instructions, such as shuffle, that require an immediate and have no corresponding register form.
+      * Checking whether the generic type parameters, if any, match those
+      supported for the current architecture. See [Supporting Generic Intrinsics](#suporting-generic-intrinsics) above.
+   * If the intrinsic fails to meet the necessary criteria:
+      * If the method is recursive, i.e. `gtIsRecursiveCall()` returns true:
+         * If the constraint is one that the JIT must expand in order to support reflection and diagnostic tools (currently only the immediate operand case), it will produce an expanded implementation (e.g. a switchtable of possible immediate values).
```

Comment:
Is it worthwhile explicitly calling out that a software implementation is undesirable (that is, an implementation that emulates the functionality, rather than a compiler fallback which still ends up executing the target instruction with the given inputs, such as via a switchtable), since it may behave differently/etc?